### PR TITLE
Adding a option to get rid of banners in your production files

### PR DESCRIPTION
**Pull request by Anbe93**
When you create a new project with this generator it will ask you if you want banners in your production files.

### CODE REVIEW

**Comment by mischah** [comment]
Related to micromata/bootstrap-kickstart#27

File: app/index.js [hunk 1]
```diff
@@ -86,6 +87,12 @@ module.exports = yeoman.Base.extend({
 					message: 'A short description of your project:'
 				},
 				{
+					type: 'confirm',
+					name: 'banner',
+					message: 'Do you like to use a banner in your production files?',
+					default: false
```
Comment (mischah) [inline app/index.js]: Wouldn’t it make sense to set the additional property `store: true` to store that value for future use of that generator?

File: app/index.js [hunk 2]
```diff
@@ -86,6 +87,13 @@ module.exports = yeoman.Base.extend({
 					message: 'A short description of your project:'
 				},
 				{
+					type: 'confirm',
+					name: 'banner',
+					message: 'Do you like to use a banner in your production files?',
```
Comment (mischah) [inline app/index.js]: Better:
```
message: 'Do you like to add comment headers containing meta information to your production files?',
```

File: test/test-app.js [hunk 3]
```diff
@@ -338,12 +343,58 @@ describe('bootstrap-kickstart with default options', function () {
...
+			'skip-install': true
+		})
+
+		// Mock the prompt answers
+		.withPrompts(prompts)
+
+		.toPromise();
+	});
+
+	it('should not have an empty string as banner within the Gruntfile', function () {
```
Comment (mischah) [inline test/test-app.js]: Better:

```
'should have the message like defined in the template of the Gruntfile'
```

**Comment by mischah** [comment]
When moving the new prompt it should be done in all phases:

+ `initializing` ➡️  `this.templateProps`
+ `prompting` ➡️  `prompts`
+ `prompting` ➡️  `this.templateProps`

**Comment by Anbe93** [comment]
Is ok to me. I just tought because it is related to the 2 Prompt obove it should be there.

File: app/index.js [hunk 4]
```diff
@@ -291,6 +291,7 @@ module.exports = yeoman.Base.extend({
 					projectHomepage: props.projectHomepage,
 					projectRepositoryType: props.projectRepositoryType,
 					projectRepository: props.projectRepository,
+					banner: config.banner,
```
Comment (mischah) [inline app/index.js]: We now have a duplicated banner property in this code block :grimacing:

See [L278](https://github.com/micromata/generator-bootstrap-kickstart/pull/39/files/f4e7ddf04983102f4b30f099aa1962b08141e728..72169d48e5d6c0245a92f38979a0b4cc689a48ab#diff-2018087f584c4398b5c3a23fc0e5f9dbL278)

**Comment by Anbe93** [comment]
I hope i did the right thing. Because i dont even know what i did over there.

**Comment by mischah** [comment]
> I hope i did the right thing. Because i dont even know what i did over there.

Hehe, Thanks. Looks good.

But we have failing tests on Travis CI now. Gonna investigate now.

**Comment by mischah** [comment]
Okay failing CI leads to an error in `app/index.js:293:14`:
https://travis-ci.org/micromata/generator-bootstrap-kickstart/jobs/168915648#L1345

And jep. There is a copy paste error in this [L293](https://github.com/micromata/generator-bootstrap-kickstart/pull/39/files#diff-2018087f584c4398b5c3a23fc0e5f9dbR293)


---

I hope the back and forth in this pull request doesn’t frustrate you.
I still have fun btw. :dancer:

**Comment by Anbe93** [comment]
I'm so excited right now. Will it work?!?!?

**Comment by mischah** [comment]
bildschirmfoto 2016-10-19 um 15 43 53

🎉

**Comment by Anbe93** [comment]
Now i need a beer 🍻

**Comment by mischah** [comment]
Here you are 🍺 

--- 

Thanks :octocat:

Your PR did it to the master branch and will be included in the next release.

**Pull request merged**